### PR TITLE
Configuration connections

### PR DESCRIPTION
**Pull request by cableman**

### CODE REVIEW

**Comment by cableman** [comment]
The last engine coding standard is due to the comment out online check, that we don't have an solution to yet.

File: engine/plugins/server/server.js [hunk 1]
```diff
@@ -66,28 +91,83 @@ module.exports = function(options, imports, register) {
             socket.emit('UpdateState', client.state);
         });
 
+        /**
+         * The first message the client should send is "ClientReady" which will validate the token and send
+         * configuration to the client base on that token.
```
Comment (tuj) [inline engine/plugins/server/server.js:96]: "base on" => "based on"

File: engine/plugins/server/server.js [hunk 2]
```diff
@@ -66,28 +91,83 @@ module.exports = function(options, imports, register) {
...
+                    socket.disconnect(true);
+                    return;
+                }
+
+                // Set that current token is valid.
+                isTokenValid = true;
+
+                // Get configuration for this client box based on config id from token validation.
+                const clientEvent = uniqId();
+                bus.on(clientEvent, (config) => {
```
Comment (tuj) [inline engine/plugins/server/server.js:113]: Is this on(clientEvent) cleaned up later? E.g. in the on("disconnect") ...

**Review by tuj — CHANGES_REQUESTED** [review]

File: engine/plugins/client/client.js [hunk 3]
```diff
@@ -23,15 +23,19 @@ const Client = function Client() {};
  * @return {*}
  *   The client for the token.
  */
-Client.prototype.load = (token) => {
+Client.prototype.load = (token, config = {}, state = {}) => {
+    // If client don't exists
```
Comment (tuj) [inline engine/plugins/client/client.js:27]: "don't" => "does not"

File: engine/plugins/server/server.js (hunk 1 above)
Comment (cableman) [reply engine/plugins/server/server.js:96]: Fixed

File: engine/plugins/client/client.js (hunk 3 above)
Comment (cableman) [reply engine/plugins/client/client.js]: Fixed

File: engine/plugins/server/server.js (hunk 2 above)
Comment (cableman) [reply engine/plugins/server/server.js]: Fixed

**Review by tuj — APPROVED** [review]

**Review by tuj — APPROVED** [review]

**Pull request merged**